src/pages/Main/FilterButton/FilterModal/FilterModal.js

```diff
@@ -0,0 +1,466 @@
+import React, { useState, useEffect } from 'react';
+import styled from 'styled-components';
+
+function Modal({ title, handleModal, tagFetchHandler }) {
+  const [filterCategories, setFilterCategories] = useState([]);
+  const [selectCategory, setselectCategory] = useState(1);
+  const [multipleAlert, setMultipleAlert] = useState(false);
+
+  useEffect(
+    () =>
+      setFilterCategories(
+        fetch('http://54.180.99.36:8000/jobpostings/tags')
+          .then(res => res.json())
+          .then(data => {
+            setFilterCategories(
+              data.result.map(({ id, name, is_multiple_choice, tags }) => ({
+                id,
+                name,
+                is_multiple_choice,
+                tags,
+                selected: [],
+              }))
+            );
+            console.log(`data`, data);
+          })
+      ),
+    []
+  );
+
+  useEffect(() => setMultipleAlert(false), [selectCategory]);
+
+  const tagListHandler = id => setselectCategory(id);
+
+  const prevSelected = id => {
+    const prevSelected = filterCategories.filter(
+      filterList => filterList.id === id
+    )[0].selected;
+    return prevSelected;
+  };
+
+  const changeTag = (id, tag) => {
+    return filterCategories.map(mapList =>
+      mapList.id === id ? { ...mapList, selected: [tag] } : mapList
+    );
+  };
+
+  const addTagList = (id, tagName, multiple) => {
+    if (multiple) {
+      setFilterCategories(
+        filterCategories.map(mapList =>
+          mapList.id === id
+            ? { ...mapList, selected: [...prevSelected(id), tagName] }
+            : mapList
+        )
+      );
+    } else if (!multiple && prevSelected(id).length === 0) {
+      setFilterCategories(changeTag(id, tagName));
+    } else {
+      setMultipleAlert(true);
+      setFilterCategories(changeTag(id, tagName));
+    }
+  };
+
+  const removeSelected = (id, tagName) => {
+    const removeTag = prevSelected(id).filter(prevTag => tagName !== prevTag);
+    setFilterCategories(
+      filterCategories.map(mapList =>
+        mapList.id === id
+          ? {
+              ...mapList,
+              selected: [...removeTag],
+            }
+          : mapList
+      )
+    );
+  };
+
+  const tagListCount = () => {
+    return filterCategories.selected
+      ? filterCategories
+          .map(setList => setList.selected.length)
+          .reduce((a, b) => a + b, null)
+      : '';
+  };
+
+  const pickTagHandler = (id, tagName, multiple) => {
+    if (prevSelected(id).includes(tagName)) removeSelected(id, tagName);
+    else if (tagListCount() === 3 && multiple)
+      alert('태그 선택은 3개만 가능 합니다.');
+    else addTagList(id, tagName, multiple);
+  };
+
+  const reset = () => {
+    setFilterCategories(
+      filterCategories.map(({ id, name, is_multiple_choice, tags }) => ({
+        id,
+        name,
+        is_multiple_choice,
+        tags,
+        selected: [],
+      }))
```

Comment:
```suggestion
      filterCategories.map(({ selected, ...rest }) => ({
        ...rest
        selected: [],
      }))
```
selected 제외하고 나머지는 그대로 들어가면 되는 것 같은데 위와 같이 표현 가능합니다!

Reply:
와! 감사합니다!!

Reply:
수정 안된 것 같습니다!